### PR TITLE
KAFKA-12829: Remove old Processor and ProcessorSupplier interfaces

### PR DESCRIPTION
**Pull request by mjsax**
Must be cherry-picked to `4.0` branch.

### CODE REVIEW

**Comment by chia7712** [comment]
@mjsax could you please fix build error?

**Comment by chia7712** [comment]
The latest commits include a trunk rebase and build fix.

**Review by chia7712 — APPROVED** [review]

**Comment by mjsax** [comment]
@chia7712 -- This PR is stacked on another one anyway, and need rebasing before we can finalize reviewing and merge it.

I hope that you pushing into my branch, does not make rebasing more difficult now... Why did you push into my branch? There was not really a good reason for it...

**Comment by chia7712** [comment]
> I hope that you pushing into my branch, does not make rebasing more difficult now... Why did you push into my branch? There was not really a good reason for it...

Apologies for the intrusion. I tried to expedite the CI process by using a merge push instead of a rebase. I assumed the merge policy is easier for you to pull the code locally (just call `git pull`). Additionally, it is easy to revert it if you don't like those commits. I won't do that again, as it causes you trouble :(

File: streams/src/main/java/org/apache/kafka/streams/processor/ProcessorContext.java [hunk 1]
```diff
@@ -105,8 +111,8 @@ void register(final StateStore store,
 
     /**
      * Schedule a periodic operation for processors. A processor may call this method during
-     * {@link Processor#init(ProcessorContext) initialization} or
-     * {@link Processor#process(Object, Object) processing} to
+     * {@link org.apache.kafka.streams.kstream.ValueTransformer#init(ProcessorContext)  initialization} or
+     * {@link org.apache.kafka.streams.kstream.ValueTransformer#transform(Object)  processing} to
```
Comment (bbejeck) [inline streams/src/main/java/org/apache/kafka/streams/processor/ProcessorContext.java:115]: Why mention `ValueTransformer`? Is this intentional?

**Review by bbejeck — APPROVED** [review]
LGTM - just one question on one of the Javadocs

File: streams/src/main/java/org/apache/kafka/streams/processor/ProcessorContext.java (hunk 1 above)
Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/processor/ProcessorContext.java:115]: Yes. `ValueTransformer` (while deprecated itself) is the last class which is still used via `KTAbel#transformValues()` which we did not yet deprecate... (There is ticket for it already: https://issues.apache.org/jira/browse/KAFKA-17178)

So we cannot remove `ValueTransformer` yet, and it make sense to refer to it here IMHO.

**Comment by mjsax** [comment]
Rebased to `trunk`. Plan to merge after CI passed if I don't hear otherwise.

**Comment by mjsax** [comment]
@chia7712 @bbejeck -- I just pushed small change -- `To` helper class was updated incorrectly, and I reverted the change. The JavaDocs for `To` are correct as-is, and don't need any update. Just an FYI.

**Comment by mjsax** [comment]
Java 23 passed.

Java 17:
```
Found 3 test failures:
FAILED ❌ RemoteLogManagerTest > testRLMOpsWhenMetadataIsNotReady()
FAILED ❌ PlaintextAdminIntegrationTest > testConsumerGroupsDeprecatedConsumerGroupState(String, String).quorum=kraft.groupProtocol=consumer
FAILED ❌ PlaintextAdminIntegrationTest > testConsumerGroups(String, String).quorum=kraft.groupProtocol=consumer
```

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `4.0` branch.

**Pull request merged**